### PR TITLE
Update RELEASE.md

### PR DESCRIPTION
**Pull request by yebai**
Add recently fixed issues and new features to release.md

### CODE REVIEW

**Comment by yebai** [comment]
@xukai92 @emilemathieu Pls take a look at this PR and feel free to add missing recently closed issues or new features.

File: RELEASE.md [hunk 1]
```diff
@@ -1,6 +1,15 @@
+# Release 0.3.1
+- Fix compatibility with Julia 0.6 [#341, #330, #293]
+- Support of Stan interface [#343, #326]
+- Fix Binomial distribution for gradients. [#311]
+- Stochastic gradient Hamiltonian Monte Carlo [#201]
+- Disable adaptive resampling for CSMC [#357]
+- Fix resampler for SMC [#338]
+- Initial implementation of Interactive PMCMC [#334]
```
Comment (emilemathieu) [inline RELEASE.md]: why initial ?

File: RELEASE.md [hunk 2]
```diff
@@ -1,6 +1,15 @@
+# Release 0.3.1
+- Fix compatibility with Julia 0.6 [#341, #330, #293]
+- Support of Stan interface [#343, #326]
+- Fix Binomial distribution for gradients. [#311]
+- Stochastic gradient Hamiltonian Monte Carlo [#201]
```
Comment (emilemathieu) [inline RELEASE.md]: There is also: Stochastic Gradient Langevin Dynamics [https://github.com/yebai/Turing.jl/pull/327]

Comment (emilemathieu) [reply RELEASE.md]: and PIMH & PMMH: https://github.com/yebai/Turing.jl/pull/364

Comment (yebai) [reply RELEASE.md]: Right - I will also add these items to RELEASE.md.

File: RELEASE.md (hunk 1 above)
Comment (yebai) [reply RELEASE.md]: perhaps I can drop this word -  since it was not included for other new samplers anyway.

File: RELEASE.md (hunk 2 above)
Comment (emilemathieu) [reply RELEASE.md]: Nice

File: RELEASE.md (hunk 1 above)
Comment (emilemathieu) [reply RELEASE.md]: 👍

**Comment by yebai** [comment]
> Couldn't we merge MH (#369) for this release?

Yes, we can now - my tagging was rejected because Julia team wants us to use 0.4 instead of 0.3.1. I will merge your PRs and make a new tagging.

**Pull request merged**